espnet2/tasks/abs_task.py

```diff
@@ -1502,18 +1502,32 @@ def build_curriculum_iter_factory(
                 drop_last=False,
                 min_batch_size=1
                 )
-
         logging.info(f"BATCH SAMPLER: {batch_sampler}")
 
-        #batches = batch_sampler.get_tasks()
-        batches = list(batch_sampler)
+        tasks = list(batch_sampler)
 
         logging.info(f"[{mode}] dataset:\n{dataset}")
         logging.info(f"[{mode}] Batch sampler: {batch_sampler}")
 
+        if iter_options.distributed:
+            world_size = torch.distributed.get_world_size()
+            rank = torch.distributed.get_rank()
+            split_tasks = []
+            for batches in tasks:
+                filtered_batches = 0
+                local_batches = []
+                for batch in batches:
+                    if len(batch) < world_size:
+                        filtered_batches += 1
+                        continue
+                    local_batches.append(batch[rank::world_size])
+                logging.warning(f"{filtered_batches} batches less than world size removed")
```

Comment:
I don't know if this is a problem since it removes batches with really long utterances.